router/functionHandler.go

```diff
@@ -217,6 +225,9 @@ func (roundTripper RetryingRoundTripper) RoundTrip(req *http.Request) (resp *htt
 				go roundTripper.funcHandler.tapService(serviceUrl)
 			}
 
+			// reset the body before every retry
+			req.Body = ioutil.NopCloser(bytes.NewReader(reqBody))
```

Comment:
Using nopcloser is a great idea!  But can we directly wrap the req body in a nopcloser instead of wrapping a buffer?

Also, what closes the request body now?  Does it never get closed?  Is that a problem?